composer/core/state.py

```diff
@@ -505,9 +505,48 @@ def load_state_dict(self, state: Dict[str, Any], strict: bool = False):
         """
         state = _ensure_backwards_compatible_checkpointing(state)
 
+        # Verify surgery algorithms from checkpoint are also enabled in current run
```

Comment:
If a comment is needed here, then this should be a helper function, e.g. something like:
```
_verify_required_algorithms_enabled
```